nidirect_contacts/src/Plugin/Block/ContactAzBlock.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\nidirect_contacts\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Routing\CurrentRouteMatch;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'ContactAzBlock' block.
+ *
+ * @Block(
+ *  id = "contact_az_block",
+ *  admin_label = @Translation("Contacts A to Z"),
+ * )
+ */
+class ContactAzBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Drupal\Core\Routing\CurrentRouteMatch definition.
+   *
+   * @var Drupal\Core\Routing\CurrentRouteMatch
+   */
+  protected $routeMatch;
+
+  /**
+   * @param array $configuration
+   *   Site configuration.
+   * @param string $plugin_id
+   *   The plugin id.
+   * @param mixed $plugin_definition
+   *   Plugin definition.
+   * @param Drupal\Core\Routing\CurrentRouteMatch $route_match
+   *   Route match object.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, CurrentRouteMatch $route_match) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->routeMatch = $route_match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $build = [];
+    $links = [];
+
+    $title = $this->t('Find contacts beginning with') . '...';
+
+    if ($this->routeMatch->getRouteName() == 'nidirect_contacts.letter') {
+      $letter = $this->routeMatch->getParameter('letter');
+      $title = $this->t('Showing entries for') . ' ' . strtoupper($letter);
```

Comment:
$letter should be passed into t() as a placeholder.

Reply:
👍